src/app/core/services/localeService.ts

```diff
@@ -1,24 +1,56 @@
-import { Injectable } from '@angular/core';
+import { Injectable, OnDestroy } from '@angular/core';
+import { BehaviorSubject, Subscription } from 'rxjs';
 import { appConfig } from 'src/app/shared/helpers/config';
+import { UserSessionService } from 'src/app/shared/services/user-session.service';
+
+export interface LanguageMismatch {
+  platformLanguage: string | undefined,
+  userDefaultLanguage: string,
+  userDefaultLanguageIsSupported: boolean,
+}
 
 @Injectable({
   providedIn: 'root'
 })
-export class LocaleService {
+export class LocaleService implements OnDestroy {
 
   readonly languages?: { tag: string, path: string }[];
   readonly currentTag?: string;
+  readonly languageMismatch$ = new BehaviorSubject<LanguageMismatch | undefined>(undefined)
 
-  constructor() {
+  private userSessionSubscription: Subscription
+
+  constructor(private userSession: UserSessionService) {
     this.languages = appConfig.languages;
     this.currentTag = this.languages?.find(l => window.location.pathname.endsWith(l.path))?.tag;
+
+    this.userSessionSubscription = this.userSession.session$.subscribe(session => {
+      if (!session) return;
+
+      const userDefaultLanguage = session.user.defaultLanguage;
+      const hasLanguageMismatch = !!this.currentTag && session.user.defaultLanguage !== this.currentTag;
+      if (hasLanguageMismatch) {
+        this.languageMismatch$.next({
+          platformLanguage: this.currentTag,
+          userDefaultLanguage,
+          userDefaultLanguageIsSupported: this.languages?.some(({ tag }) => tag === userDefaultLanguage) ?? false,
+        });
+      }
+    });
+  }
```

Comment:
I can't do that for now because the user session is not refreshed causing the dialog to stay. Currently I can close the dialog by explicitly calling `languageMismatch$.next()`, which a smell too I guess.

Shouldn't I make the refresh task first and then come back to this one ?